fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_HoldAtPage.js

```diff
@@ -55,22 +55,112 @@ class CDUHoldAtPage {
+                                    ret = Math.abs(a.magneticCourse - inboundMagneticCourse) - Math.abs(b.magneticCourse - inboundMagneticCourse);
+                                }
+                                return ret;
+                            });
+
+                        if (holds[0]) {
+                            defaultHold = {
+                                inboundMagneticCourse: holds[0].magneticCourse,
+                                turnDirection: holds[0].turnDirection,
+                                time: holds[0].lengthTime ? holds[0].lengthTime : undefined,
+                                distance: holds[0].length ? holds[0].length : undefined,
+                                type: HoldType.Database,
+                            };
+                        }
+
+                        CDUHoldAtPage.addOrEditManualHold(
+                            mcdu,
+                            waypointIndexFP,
+                            // eslint-disable-next-line prefer-object-spread
+                            Object.assign({}, defaultHold),
+                            modifiedHold,
+                            defaultHold,
+                            forPlan,
+                            inAlternate,
+                        );
+                    }).catch(() => {
+                        CDUHoldAtPage.addOrEditManualHold(
+                            mcdu,
+                            waypointIndexFP,
+                            // eslint-disable-next-line prefer-object-spread
+                            Object.assign({}, defaultHold),
+                            modifiedHold,
+                            defaultHold,
+                            forPlan,
+                            inAlternate,
+                        );
+                    });
+                    return;
+                }
             }
 
-            mcdu.flightPlanService.addOrEditManualHold(
+            CDUHoldAtPage.addOrEditManualHold(
+                mcdu,
                 waypointIndexFP,
                 // eslint-disable-next-line prefer-object-spread
                 Object.assign({}, defaultHold),
                 modifiedHold,
                 defaultHold,
                 forPlan,
                 inAlternate,
-            ).then((holdIndex) => {
-                CDUHoldAtPage.DrawPage(mcdu, holdIndex, waypointIndexFP, forPlan, inAlternate);
-            });
+            );
+        }
+    }
+
+    static holdVerticalDistanceFromAlt(altitude, hold) {
+        switch (hold.altitudeDescriptor) {
+            case "B":
+                if (altitude <= hold.altitude1 && altitude >= hold.altitude2) {
+                    return 0;
+                }
+                if (altitude > hold.altitude1) {
+                    return altitude - hold.altitude1;
+                }
+                return hold.altitude2 - altitude;
+            case "+":
+                return altitude >= hold.altitude1 ? 0 : hold.altitude1 - altitude;
```

Comment:
If you want, you could do
```suggestion
                return Math.max(0, hold.altitude1 - altitude);
```
The same for the next case.